### PR TITLE
[DOCS] Avoid trailing newline in apikey base64 encoding

### PR DESCRIPTION
**Pull request by jubr**
Hi there, I just ran into forgetting not to add the newline in generating the base64 encoded value for the ApiKey. Thought it might be nice to include this in the official docs. Feel free to reformat the content if it does not fit the guidelines exactly.

### CODE REVIEW

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks for this contribution @jubr. I moved around the text so the encoding step precedes the request.

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jrodewig** [comment]
@ywangd Do you mind taking a look to ensure I didn't miss anything here? Thanks.

File: x-pack/docs/en/rest-api/security/create-api-keys.asciidoc [hunk 1]
```diff
@@ -78,7 +78,9 @@ expire.
 [[security-api-create-api-key-example]]
 ==== {api-examples-title}
 
-The following example creates an API key:
+The following example creates an API key.
+
+NOTE: If your node has `xpack.security.http.ssl.enabled` set to `true`, then you must specify `https` when creating your API key.
```
Comment (ywangd) [inline x-pack/docs/en/rest-api/security/create-api-keys.asciidoc:83]: Even though this is true, there is no where we use `http` or `https` explicitly in this page to create API keys. So this note feels out of place. I think we could revert it back to its original place and change it to something like:

```
If your node has `xpack.security.http.ssl.enabled` set to `true`, then you must specify `https` in the following command.
```

File: x-pack/docs/en/rest-api/security/create-api-keys.asciidoc [hunk 2]
```diff
@@ -132,14 +134,20 @@ API key information.
...
+To create your credentials, encode the returned `id:api_key` in base64. Ensure
+the value does not include a trailing newline.
 
-NOTE: If your node has `xpack.security.http.ssl.enabled` set to `true`, then you must specify `https` when creating your API key.
+[source,shell]
+--------------------------------------------------
+echo -n "VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw" | base64
+# returns VnVhQ2ZHY0JDZGJrUW0tZTVhT3g6dWkybHAyYXhUTm1zeWFrdzl0dk5udw==
+--------------------------------------------------
+// NOTCONSOLE
```
Comment (ywangd) [inline x-pack/docs/en/rest-api/security/create-api-keys.asciidoc:145]: Tests failed and suggested to remove this line
> en/rest-api/security/create-api-keys.asciidoc[141:144](shell)// NOTCONSOLE: No need for NOTCONSOLE if snippet doesn't contain `curl`.

File: x-pack/docs/en/rest-api/security/create-api-keys.asciidoc [hunk 3]
```diff
@@ -132,14 +134,20 @@ API key information.
 <2> optional expiration in milliseconds for this API key
 <3> generated API key
 
-The API key returned by this API can then be used by sending a request with an
-`Authorization` header with a value having the prefix `ApiKey` followed
-by the _credentials_, where _credentials_ are the base64 encoding of `id` and `api_key` joined by a colon.
+To create your credentials, encode the returned `id:api_key` in base64. Ensure
+the value does not include a trailing newline.
```
Comment (ywangd) [inline x-pack/docs/en/rest-api/security/create-api-keys.asciidoc:138]: I think where `id:api_key` comes from is not clear. Also the term "credentails" is introduced with no context. Overall I prefer the original version. 

I think what we really want to say here is just the fact: when creating the credentials using `echo`, we must be careful to not include a trailing newline (echo's default behaviour). This is just one way of creating the credentials and we should be explicit about it. So I'd suggest we keep the original text and add a new sentence like:
```
One way to create the _credentails_ from CLI on an Unix-like system is as the follows:
[source,shell]
--------------------------------------------------
# Please note the use of "-n" to instruct echo to not print the trailing newline character.
# It should return VnVhQ2ZHY0JDZGJrUW0tZTVhT3g6dWkybHAyYXhUTm1zeWFrdzl0dk5udw==
echo -n "VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw" | base64
--------------------------------------------------
```

**Review by ywangd — COMMENTED** [review]
I left a few comments.

**Comment by jrodewig** [comment]
Thanks for the feedback @ywangd. I've reverted my changes and implemented your suggestions.

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jubr** [comment]
Nice discussion @jrodewig and @ywangd! Resulting in a minimal and clean 👉 in the right direction for users to circumvent this pitfall. Tnx!

**Pull request merged**